juice/src/net/loss/negative_log_likelihood.rs

```diff
@@ -55,15 +45,17 @@ impl<B: IBackend> Layer<B> for NegativeLogLikelihood {
         let native_labels = labels_data.read(native.device()).unwrap().as_slice::<f32>();
         let mut writable_gradient = vec![0f32; probabilities_gradient.borrow().desc().size()];
 
+        let num_classes = self.descriptor.input(0).unit_shape().size();
+
         for (batch_n, &label_value) in native_labels.iter().enumerate() {
-            let index = (self.num_classes * batch_n) + label_value as usize;
+            let label_index = label_value as usize;
+            assert!(label_index < num_classes, "Wrong label {} exceeding input size {}", label_index, num_classes);
+
+            let index = (num_classes * batch_n) + label_index;
             writable_gradient[index] = -1f32;
         }
         crate::util::write_to_memory(
-            probabilities_gradient
-                .borrow_mut()
-                .write_only(native.device())
-                .unwrap(),
+            probabilities_gradient.borrow_mut().write_only(native.device()).unwrap(),
```

Comment:
I am still torn if we should have these unwraps or rather make it an error and have the API have a result.
We disussed it briefly in another PR, and it's - again - out of scope for this, but it might be worth
looking into it and writing down the resolution so I don't brig it up again.

Reply:
Sure, I can do this in the next PR. Filed #185 to track.

Reply:
Thank you ❤️